### PR TITLE
JLayout to render debug backtrace; Clickable links for XDebug

### PR DESCRIPTION
**Pull request by n9iels**
This Pull Request is a redo of #3780, see that PR for more information
## Modifications

This PR adds a JLayout file for the callstack backtrace. The debug plugin and isis template include this layout now.
On request of @roland-d in the original PR I also add clickable links for XDebug session. If the flag `xdebug.file_link_format` is set in the php.ini the location link will be clickable.
## Test instructions
1. Go to the Global Configuration, system tab and enable "Debug system"
2. Type `/administrator/index.php?option=foo` in the browser and you will see the following error page
   isis-error
3. If you have XDebug you will see the following error page
   isis-error-xdebug

### CODE REVIEW

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on d475a042a2408e2fbcd6d3a19f29b78e0ceaafa4<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10306">issues.joomla.org/joomla-cms/10306</a>.</sub>

**Comment by wilsonge** [comment]
Sorry to be a pain - but rather than putting this in the global JLayout folder can we please make it extension specific like we did for the stats layouts (https://github.com/joomla/joomla-cms/tree/staging/plugins/system/stats) - you'll need to adapt the code slightly like @phproberto did at https://github.com/joomla/joomla-cms/blob/staging/plugins/system/stats/stats.php#L279-L312

**Comment by andrepereiradasilva** [comment]
yes @wilsonge is right. Since this is only used when the debug plugin is enabled it should be a layout only of the debug plugin.

**Comment by mbabker** [comment]
> Sorry to be a pain - but rather than putting this in the global JLayout folder can we please make it extension specific like we did for the stats layouts

Please don't.  @wilsonge `JDocumentError` is also using this.  And at a quick glance it looks like `JError::renderBacktrace()` could too (I know that one's deprecated but proxying that to "current" code at least keeps things consistent).

**Comment by brianteeman** [comment]
Obviously this needs new unit tests

**Comment by n9iels** [comment]
I don't have experience with unit testing :sweat_smile:
So if someone else can correct these errors, makes a PR in my branch (https://github.com/n9iels/joomla-cms/tree/backtrace-isis)...

File: libraries/joomla/document/error.php [hunk 1]
```diff
@@ -156,61 +156,13 @@ public function _loadTemplate($directory, $filename)
 	 */
 	public function renderBacktrace()
 	{
-		// If no error object is set return null
```
Comment (mbabker) [inline libraries/joomla/document/error.php]: This check needs to be restored.  It is in part what's causing your unit test failures.

**Comment by n9iels** [comment]
Better late than never!
I restored the checks as @mbabker suggested, travis seems to be happy now.

**Comment by mbabker** [comment]
If it's not too hard can you also have JError::renderBacktrace() use this
layout?  If it is, no big deal, I'd just like old and new APIs to be
consistent if possible.

Otherwise this looks good to me.

**Comment by n9iels** [comment]
I have some problems with this method in the JLayout: https://github.com/n9iels/joomla-cms/blob/d55866ca9ede1aff6652711f44f58732f09c7405/layouts/joomla/error/backtrace.php#L34

It is declared in the plugin class, but obviously the JLayout cannot find it. What is pretty why to handle this? Creating is `PlgSystemDebugHelper`?

@mbabker `JError` is deprecated so we don't have to support it. But I can give it a try if I solved the problem above :-)

**Comment by mbabker** [comment]
Yes, you'll need a helper method.  It should be in the libraries though, not the debug plugin.  You could probably add a method to `JHtml` for this.

As for `JError` being deprecated, well aware of that, however it's also good to proxy the deprecated code to new code to ease the transition where it makes sense.  This is one of those places where in theory it should be easy to change and it results in one less inconsistency across our overall API (merging three places rendering in essence the same markup into one source is always good).

**Comment by mbabker** [comment]
https://github.com/joomla/joomla-cms/pull/14957 implemented this.

**Pull request closed without merging**